src/java/org/jivesoftware/openfire/muc/spi/MUCPersistenceManager.java

```diff
@@ -129,8 +129,8 @@
     private static final String DELETE_USER_MUCAFFILIATION =
         "DELETE FROM ofMucAffiliation WHERE jid=?";
     private static final String ADD_CONVERSATION_LOG =
-        "INSERT INTO ofMucConversationLog (roomID,sender,nickname,logTime,subject,body,stanza) " +
-        "VALUES (?,?,?,?,?,?,?)";
+        "INSERT INTO ofMucConversationLog (roomID,messageID,sender,nickname,logTime,subject,body,stanza) " +
+        "SELECT ?,COUNT(*),?,?,?,?,?,? FROM ofMucConversationLog";

```

Comment:
Those are all really database-specific, and often really hard to add to an existing table. I'd have had to do the upgrade by creating a new table and populating it from the old one (which I can do, but getting it right in every database we support....?)